### PR TITLE
Adding code coverage workflow

### PR DESCRIPTION
**Pull request by gvegayon**
- Adds coverage report via `pytest-cov`.
- Sets up codecov.io and uploads report.
- Adds a badge to README.md
- Renames existing workflows to have shorter badge names.

### CODE REVIEW

**Review by dylanhmorris — APPROVED** [review]
I somewhat preferred the more verbose `test_model.yaml` and `test_pipeline.yaml` names for the workflow files, but I am open to this change, and I won't block this on that account alone. Everything else looks good.

**Comment by gvegayon** [comment]
> I somewhat preferred the more verbose `test_model.yaml` and `test_pipeline.yaml` names for the workflow files, but I am open to this change, and I won't block this on that account alone. Everything else looks good.

No strong opinion on that, so I'll revert it before merge.

**Review by dylanhmorris — APPROVED** [review]

**Pull request merged**
